src/ImageReconstruction.jl

```diff
@@ -60,41 +50,58 @@ function _ramp_spatial(N::Int, τ)
     h
 end
 
-_zero_pad(p::AbstractVector, N::Int) = vcat(p, zeros(N))
-
 """
-Inverse radon transform using a ramp filter and a pixel-driven algorithm.
+	iradon(P::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
+
+Inverse radon transform of a sinogram `P` with view angles `θ` in radians and
+detector sampling `t` producing an image on a 128x128 matrix.
 """
-function iradon(sinogram::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
+function iradon(P::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
     pixels = 128
 
     N = length(t)
     K = length(θ)
     Npad = nextpow(2, 2 * N - 1)
     τ = step(t)
-    ramp = fft(_zero_pad(_ramp_spatial(N, τ), Npad - N))
-    i = div(N, 2) + 1
+    ramp = fft(_ramp_spatial(N, τ, Npad))
+    i = N ÷ 2 + 1
     j = i + N - 1
 
-    image = zeros(eltype(sinogram), pixels, pixels)
-    Q = Vector{eltype(sinogram)}(undef, N)
-    for (k, θₖ) in enumerate(θ)
+    T = eltype(P)
+    image = [zeros(T, pixels, pixels) for _ = 1:nthreads()]
+    P′ = [Vector{Complex{T}}(undef, Npad) for _ = 1:nthreads()]
+    Q = [Vector{T}(undef, N) for _ = 1:nthreads()]
+
+    l = SpinLock()
+    @inbounds @threads for (k, θₖ) in collect(enumerate(θ))
+        id = threadid()
+
         # filter projection
-        Q[:] .= τ .* real.(ifft(fft(_zero_pad(sinogram[:, k], Npad - N)) .* ramp)[i:j])
+        P′[id][1:N] .= P[:, k]
+        P′[id][N+1:end] .= 0
+
+        # Need to prevent multiple thread execution during fft/ifft.
+        # double free occurs otherwise.
+        # https://github.com/JuliaMath/FFTW.jl/issues/134
+        lock(l)
+        fft!(P′[id])
+        P′[id] .*= ramp
+        ifft!(P′[id])
+        unlock(l)
+        Q[id] .= τ .* real.(@view P′[id][i:j])
+
+        Qₖ = LinearInterpolation(t, Q[id])
 
         # backproject
-        for c in CartesianIndices(image)
+        for c in CartesianIndices(first(image))
             x = c.I[2] - pixels ÷ 2 + 0.5
             y = c.I[1] - pixels ÷ 2 + 0.5
+            x^2 + y^2 ≥ pixels^2 / 4 && continue
             t′ = x * cos(θₖ) + y * sin(θₖ)
-            # linear interpolation
-            #   image pixel (x, y) at θₖ is projected on t[z] and t[z+1]
-            z = convert(Int, round((t′ - minimum(t)) / step(t) + 1))
-            α = abs(t′ - t[z])
-            image[c] += (1 - α) * Q[z] + α * Q[z + 1]
+            image[id][c] += Qₖ(t′)
```

Comment:
Check to see if hoisting `imageid = image[id]` out of the loop improves performance.